src/main/java/com/example/kidz_school/service/impl/ConfirmationTokenServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.example.kidz_school.service.impl;
+
+import com.example.kidz_school.config.properties.ConfirmationTokenConfigProperties;
+import com.example.kidz_school.entity.ConfirmationToken;
+import com.example.kidz_school.entity.User;
+import com.example.kidz_school.repository.ConfirmationTokenRepository;
+import com.example.kidz_school.service.ConfirmationTokenService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class ConfirmationTokenServiceImpl implements ConfirmationTokenService {
+    private final ConfirmationTokenRepository confirmationTokenRepository;
+
+    //?????????????????????@Schedulllllllllllllllllllll
+    public void deleteExpiredTokens() {
+        LocalDateTime currentDateTime = LocalDateTime.now();
+        List<ConfirmationToken> expiredTokens = confirmationTokenRepository.findByExpiresAtBefore(currentDateTime);
+
+        confirmationTokenRepository.deleteAll(expiredTokens);
+    }
+
+    @Override
+    public ConfirmationToken save(ConfirmationToken confirmationToken) {
+        return confirmationTokenRepository.save(confirmationToken);
+    }
+
+    @Override
+    public ConfirmationToken findByToken(String token) {
+        return confirmationTokenRepository.findByToken(token);
+    }
+
+    @Override
+    public ConfirmationToken generateConfirmationToken(UUID uuid, User savedUser) {
+        ConfirmationToken token = new ConfirmationToken();
+        long expiration = new ConfirmationTokenConfigProperties().getExpiration();
+        token.setExpiresAt(LocalDateTime.now().plusMinutes(expiration));
```

Comment:
How can you instantiate the ConfirmationTokenConfigProperties class in the method. It's a Component Hasmik jan. You should inject the class as dependency.